splitn/main.py

```diff
@@ -0,0 +1,85 @@
+from itertools import chain, combinations, repeat
+from collections.abc import Generator
+from copy import deepcopy
+from random import randint
+from re import compile
+
+import typer
+
+class CombinationsGenerator():
```

Comment:
Did you think about using loose, independent functions instead of packing them into a class?